baybe/surrogates/base.py

```diff
@@ -51,14 +51,6 @@ class Surrogate(ABC, SerialMixin):
     """Class variable encoding whether or not the surrogate supports transfer
     learning."""
 
-    # Object variables
-    # TODO: In a next refactoring, the user friendliness could be improved by directly
-    #   exposing the individual model parameters via the constructor, instead of
```

Comment:
Will this change (e.g. exposing the individual model parameters via the constructor) be implemented later? I do not see this being implemented in this PR, so to me it feels like this PR currently just makes it impossible for users to specify their models, actually making our code less flexible and user friendly.

Reply:
Hi @AVHopp, we need to distinguish two things here:

1. The first is the possibility to provide `model_params` as such (whether as dict or as individual exposed attributes doesn't matter in this context). Note that this has **NOT** been removed per se, but has only been dropped from the base class since its actually not a common attribute of all subclasses. Having it in the base class resulted in a very weird class architecture, i.e. we offered it to all surrogates but then had to i) manually forbid it again in the custom model and ii) not even use it in the GP. So instead of offering and removing/not using, we now simply expose it only where really needed. That is, you can still pass your `model_params` to all surrogates other than GP/custom – they have defined this attribute locally in their class body. Does the explanation make it more clear?

2. The comment was about the idea of improving user experience by not just exposing a dict but exposing the actual possible parameters in the constructor. That would be nicer because otherwise it's really not clear for people what they are actually allowed to pass without looking at the docs of the respective other packages. Nothing speaks against making this change in the future (there are in fact several ways how we can achieve this, both explicitly and dynamically) but I simply dropped the comment here because the referenced attributed `model_params` was dropped here

Reply:
Makes sense, thanks for explaining :)